trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/rest/TrinoS3ProxyClient.java

```diff
@@ -118,7 +119,11 @@ public void proxyRequest(SigningMetadata signingMetadata, ParsedS3Request reques
                 .session()
                 .ifPresent(sessionToken -> remoteRequestHeaders.add("x-amz-security-token", sessionToken));
 
-        request.entitySupplier().ifPresent(entitySupplier -> {
+        signingMetadata.entity().ifPresentOrElse(metadataEntity -> {
+            remoteRequestBuilder.setBodyGenerator(createStaticBodyGenerator(metadataEntity));
```

Comment:
I feel that this may be the wrong place to have this - some of this logic is duplicated from the `SigningController`.

If `encoding` is:
- `chunked`: We'll create a `StreamingBodyGenerator(entityStream)` usign the entityStream in the request
- `aws-chunked`: we'll wrap the `entityStream` in the request inside a `ChunkedInputStream`
- Anything else: we're loading this from the `signingMetadata`

Regardless of what the encoding is, we always set the hash header to `UNSIGNED-PAYLOAD`

If (as per my other comment in `SigningMetadata`) we have a generic `bytes` provider in the request, we could merge all these cases. What do you think?

Also - should the `content-length` and `decoded-content-length` be included in the request for chunked requests? [Docs](https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-streaming.html#example-signature-calculations-streaming)

Reply:
See my comments. Chunked upload verification/signing needs to be handled. It's complex and Minio kept complaining so I gave up for now, but still want to do it. Once some of the remaining PRs are merged we can do this: https://github.com/trinodb/s3-proxy/issues/66

Reply:
The latest push removes organizes the request content management better. LMKWYT

Reply:
Yup this looks much nicer! Thanks for the change!